s3bucket/main.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_s3_bucket" "codedeploy_bucket" {
+  bucket = "${var.name_prefix}-codedeploy-releases"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_iam_policy" "codedeploy_policy" {
```

Comment:
because our module creates the policy.  That is why it makes it hard to reuse it